### DISCUSSION TITLE
Welcome to FastForward Discussions!

### DISCUSSION THREAD

**Discussion by NotAProton** [FastForwardTeam/FastForward#74] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together .

**Comment by Jafrazar** [comment]
Does it work on Brave? I'm ignorant, sorry

**Reply by NotAProton** [reply]
Chrome, Edge & Brave are all based on Chromium. Works on all 3

**Comment by thatguy1943** [comment]
Hello! I have a few questions. I am not very familiar with javascript at all. I did take a look at the injection script to see if I could make out what what did and how it worked, but ended up pretty much where I started. I am familiar with M code in Power Query, but I don't know if that would help me in understanding the code that is presented here. I know they are different, but I would assume they are similar in the sense that once you know what certain "words", you can basically figure out the rest.

How does this work exactly? Is there a table that contains urls and the place it leads to? Does it work by taking information from the page once it is loaded? The reason I am asking is because I have encountered urls that need to be redirected 4 or 5 times before getting to the content. Here's an example:

Start URL: pastebin1 > linkvertise1 > pastebin2 > bit.ly1 > justetext1 > linkvertise2 > pastebin3 > linkvertise3 > Final URL: pastebin4

The script work great, but as you can see, I have to keep copy and pasting new urls.

Would there be a possible way to have an option to add in a "Final Destination" button or something so that the next person would just have to go "pastebin1 > linkvertise1 > pastebin4" ? I know there is a security concern here, but maybe have it meet a certain number of entries? After x number of people confirm the same "Final Destination", make it redirect to it automatically? 

I don't know, just brainstorming here.

**Reply by thatguy1943** [reply]
Maybe add in an optional feature where if the redirect leads to a "pastebin" site, have a pop up that asks "Is there another redirect?" and a field to input that next link. Or have it pop up in the corner after every redirect and an option in the settings to turn it off/set how long it stays up after the redirect. If no action is taken after the set amount of time, it assume that it is indeed the final link.

**Reply by NotAProton** [reply]
How most sites are bypassed is we devs write small bits of javascript which are injected into the site. Here's a tutorial if want to write some yourself https://gist.github.com/NotAProton/063446f39d6715e9ee79a8a8b8709cd4

We try not to do too much cross-site stuff because security

**Reply by thatguy1943** [reply]
I figured as much.

When I get to ifElement(), can I select by the class of an element? So instead of "a#id[href]" I would do "a.class[href]". I'm just assuming since uBlockOrigin uses "#" for IDs as well and "." for classes.

Also, when viewing the page source, what is the process for getting the url from within a script? I'll have to find another link to the site, but when I viewed the source, it was not in the "a.skip-btn" element, but instead within a script. It was something like this:

```
<script>
        window.addEventListener('someTerm', () => {
            const dontSkip = select('.no-skip')
            var time = '30'
            var n = someFunction(time);
            for() {
                    countdown function or something
            }

            (function(){
                setTimeout(()=>{
                    $('.skip-btn').attr('href','http://dutchycorp.ovh/sl/DmtIL8Om?verif=kVW6TL0r')
                    dontSkip.remove()
                }, some math done here)
            })()
       I don't remember the rest as I didn't pay attention to it
</script>
```

**Comment by itspro-dev** [comment]
Gplink links buypass possible?

**Comment by thatguy1943** [comment]
Not sure if you could see the reply I left a few days ago but I have another issue with the coding. How would I bypass this?

```
<div id="wpsafe-link">
	<a onclick="window.open('https://google.com/redirect_link', '_self')" rel="nofollow" style="cursor:pointer"></a>
</div>
```
This is what I tried adding into the extension
```
domainBypass("fssquad.com", () => {
	ensureDomLoaded(() => {
		ifElement("div#wpsafe-link", d => {
			safelyNavigate(d.getElementsByTagName("a")[0].onclick)
		})
	})
})
```

EDIT: I needed to add "()" after "onclick" as so:

```
domainBypass("fssquad.com", () => {
	ensureDomLoaded(() => {
		ifElement("div#wpsafe-link", d => {
			safelyNavigate(d.getElementsByTagName("a")[0].onclick())
		})
	})
})
```

Still wondering about the previous comment though

**Comment by itspro-dev** [comment]
Contact me with email : ***@***.***
